src/App.tsx

```diff
@@ -1,16 +1,50 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { UserWarning } from './UserWarning';
 import { USER_ID } from './api/todos';
 
+import { getTodos } from '../src/api/todos';
+import { Todo } from '../src/types/Todo';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string | null>(null);
+  const [isErrorVisible, setIsErrorVisible] = useState(true);
+  const [filter, setFilter] = useState<'all' | 'active' | 'completed'>('all');
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      try {
+        setIsLoading(true);
+        const todosData = await getTodos();
+        setTodos(todosData);
+        setError(null);
+      } catch (err) {
+        setError('Unable to load todos');
+        console.error('Unable to load todos', err);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    loadTodos();
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const filteredTodos = todos.filter(todo => {
+    if (filter === 'active') return !todo.completed;
+    if (filter === 'completed') return todo.completed;
+    return true;
+  });
+
   return (
     <div className="todoapp">
+      {error && <div className='error' data-cy='ErrorNotification'>{error}</div>}
```

Comment:
There is ErrorNotification below - use it to display the correct error and remove this div here